File: code_examples/4-Functions/4-Functions_with_Multiple_Parameters/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module calculates area of a rectangle
"""


def calculate_area(length: int, width: int) -> int:
    """The calculate_area() function calculates area of a rectangle, returns int."""

    # process
    area = length * width

    # output
    return area


def main() -> None:
    """The main() function just calls other functions, returns None."""

    # input
    length_from_user = int(input("Enter the length of a rectangle (cm): "))
    width_from_user = int(input("Enter the width of a rectangle (cm): "))
    print("")

    # call functions
    area = calculate_area(width = width_from_user, length = length_from_user)

    # output
    print(f"The area is {area} cm²")

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/4-Functions/4-Functions_with_Multiple_Parameters/Java/Main.java
/*
 * This program calculates area of rectangle
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

final class Main {
  /**
   * Calculates area of rectangle.
   *
   * @param args nothing passed in
   */
  public static int calculateArea(int length, int width) {
    // process
    int area = length * width;

    return area;
  }

  private Main() {
    // Prevent instantiation
    // Optional: throw an exception e.g. AssertionError
    // if this ever *is* called
    throw new IllegalStateException("Cannot be instantiated");
  }

  /**
   * Main entry point into program.
   *
   * @param args nothing passed in
   */
  public static void main(final String[] args) {
    // this function does the input and output
    int area = 0;

    // input
    Scanner scanner = new Scanner(System.in);
    System.out.print("Enter the length of a rectangle (cm): ");
    int lengthFromUser = scanner.nextInt();
    System.out.print("Enter the width of a rectangle (cm): ");
    int widthFromUser = scanner.nextInt();
    System.out.println();

    // call functions
    area = calculateArea(lengthFromUser, widthFromUser);

    // output
    System.out.printf("The area is %d cm²%n", area);

    System.out.println("\nDone.");
  }
}
